ext/reflection/php_reflection.c

```diff
@@ -2690,6 +2761,91 @@ ZEND_METHOD(reflection_parameter, isVariadic)
 }
 /* }}} */
 
+/* {{{ proto public bool ReflectionType::allowsNull()
+  Returns whether parameter MAY be null */
+ZEND_METHOD(reflection_type, allowsNull)
+{
+	reflection_object *intern;
+	type_reference *param;
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+	GET_REFLECTION_OBJECT_PTR(param);
+
+	RETVAL_BOOL(param->arg_info->allow_null);
+}
+/* }}} */
+
+/* {{{ proto public bool ReflectionType::isClassOrInterface()
+  Returns whether parameter MUST be a class/interface instance */
+ZEND_METHOD(reflection_type, isClassOrInterface)
+{
+	reflection_object *intern;
+	type_reference *param;
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+	GET_REFLECTION_OBJECT_PTR(param);
+
+	RETVAL_BOOL(param->arg_info->type_hint == IS_OBJECT);
+}
+/* }}} */
+
+/* {{{ _reflection_type_get_class_name */
+static zend_string *_reflection_type_get_class_name(zend_function *fptr, zend_arg_info *arg_info)
+{
+	zend_string *class_name = NULL;
+	const char *name;
+	size_t len;
+
+	if (fptr->type == ZEND_INTERNAL_FUNCTION) {
+		name = ((zend_internal_arg_info*)arg_info)->class_name;
+		len = strlen(name);
+	} else {
+		class_name = arg_info->class_name;
+		name = class_name->val;
+		len = class_name->len;
+	}
+
+	if (fptr->common.scope) {
+		if (len == sizeof("self") - 1 && !strncasecmp(name, "self", sizeof("self") - 1)) {
+			class_name = fptr->common.scope->name;
+		} else if (len == sizeof("parent") - 1 && !strncasecmp(name, "parent", sizeof("parent") - 1)) {
+			class_name = fptr->common.scope->parent->name;
```

Comment:
oh yes, I was unsure if it could happen and I was supposed to ask, but I forgot. I will add it soon.